modules/flex/metadata.yaml

```diff
@@ -0,0 +1,148 @@
+        varType: string
+        defaultValue: default
+      - name: launcher_machine_type
+        description: The machine type to use for launching the job.
+        varType: string
+        defaultValue: ""
+      - name: machine_type
+        description: The machine type to use for the job.
+        varType: string
+        defaultValue: ""
+      - name: sdk_container_image
+        description: Docker registry location of container image to use for the 'worker harness. Default is the container for the version of the SDK. Note this field is only valid for portable pipelines.
+        varType: string
+      - name: use_public_ips
+        description: Specifies whether Dataflow workers use external IP addresses. If the value is set to false, Dataflow workers use internal IP addresses for all communication.
+        varType: bool
+        defaultValue: false
+      - name: enable_streaming_engine
+        description: Enable/disable the use of Streaming Engine for the job.
+        varType: bool
+        defaultValue: false
+      - name: autoscaling_algorithm
+        description: The algorithm to use for autoscaling.
+        varType: string
+      - name: skip_wait_on_job_termination
+        description: If set to true, terraform will treat DRAINING and CANCELLING as terminal states when deleting the resource, and will remove the resource from terraform state and move on.
+        varType: bool
+        defaultValue: false
+      - name: kms_key_name
+        description: "The name for the Cloud KMS key for the job. Key format is: projects/PROJECT_ID/locations/LOCATION/keyRings/KEY_RING/cryptoKeys/KEY"
+        varType: string
+      - name: additional_experiments
+        description: List of experiments that should be used by the job. An example value is `['enable_stackdriver_agent_metrics']`
+        varType: list(string)
+        defaultValue: []
+      - name: parameters
+        description: Key/Value pairs to be passed to the Dataflow job (as used in the template).
+        varType: map(string)
+        defaultValue: {}
+      - name: labels
+        description: User labels to be specified for the job.
+        varType: map(string)
+        defaultValue: {}
+    outputs:
+      - name: container_spec_gcs_path
+        description: The GCS path to the Dataflow job Flex Template.
+        type: string
+      - name: id
+        description: The unique Id of the newly created Dataflow job
+        type: string
+      - name: name
+        description: The name of the dataflow job
+        type: string
+      - name: state
+        description: The state of the newly created Dataflow job
+        type: string
+      - name: temp_location
+        description: The Cloud Storage path to use for temporary files. Must be a valid Cloud Storage URL, beginning with gs://.
+        type: string
+  requirements:
+    roles:
+      - level: Project
+        roles:
+          - roles/owner
```

Comment:
The roles are coming from https://github.com/terraform-google-modules/terraform-google-dataflow/blob/main/test/setup/iam.tf#L19.

`roles/owner` is high-privilege role. Is this needed? Could you update this to have only minimal set of roles. Something like https://github.com/terraform-google-modules/terraform-google-memorystore/blob/main/metadata.yaml#L206